File: KempIS Frontend/src/app/core/auth/roles.ts
/** Mirrors `Application/Abstractions/Authentication/Roles.cs` on the
 *  backend - keep in sync. */
export const Roles = {
  Guest: "Guest",
  Receptionist: "Receptionist",
  Accountant: "Accountant",
  CleaningStaff: "CleaningStaff",
  Manager: "Manager",
} as const;

export type RoleName = (typeof Roles)[keyof typeof Roles];

File: KempIS Backend/src/Application/Abstractions/Authentication/Roles.cs
namespace Application.Abstractions.Authentication;

public static class Roles
{
  public const string Guest = nameof(Guest);
  public const string Receptionist = nameof(Receptionist);
  public const string Accountant = nameof(Accountant);
  public const string CleaningStaff = nameof(CleaningStaff);
  public const string Manager = nameof(Manager);

  public static readonly IReadOnlyList<string> All =
    [Guest, Receptionist, Accountant, CleaningStaff, Manager];
}
